apis/compute/v1beta1/gkecluster_types.go

```diff
@@ -0,0 +1,1260 @@
+	return nil
+}
+
+// GKEClusterObservation is used to show the observed state of the GKE cluster resource on GCP.
+type GKEClusterObservation struct {
+	// Conditions: Which conditions caused the current cluster state.
+	Conditions []*StatusCondition `json:"conditions,omitempty"`
+
+	// CreateTime: [Output only] The time the cluster was created,
+	// in
+	// [RFC3339](https://www.ietf.org/rfc/rfc3339.txt) text format.
+	CreateTime string `json:"createTime,omitempty"`
+
+	// CurrentMasterVersion: [Output only] The current software version of
+	// the master endpoint.
+	CurrentMasterVersion string `json:"currentMasterVersion,omitempty"`
+
+	// CurrentNodeCount: [Output only]  The number of nodes currently in the
+	// cluster. Deprecated.
+	// Call Kubernetes API directly to retrieve node information.
+	CurrentNodeCount int64 `json:"currentNodeCount,omitempty"`
+
+	// CurrentNodeVersion: [Output only] Deprecated,
+	// use
+	// [NodePools.version](/kubernetes-engine/docs/reference/rest/v1/proj
+	// ects.zones.clusters.nodePools)
+	// instead. The current version of the node software components. If they
+	// are
+	// currently at multiple versions because they're in the process of
+	// being
+	// upgraded, this reflects the minimum version of all nodes.
+	CurrentNodeVersion string `json:"currentNodeVersion,omitempty"`
+
+	// Endpoint: [Output only] The IP address of this cluster's master
+	// endpoint.
+	// The endpoint can be accessed from the internet
+	// at
+	// `https://username:password@endpoint/`.
+	//
+	// See the `masterAuth` property of this resource for username
+	// and
+	// password information.
+	Endpoint string `json:"endpoint,omitempty"`
+
+	// ExpireTime: [Output only] The time the cluster will be
+	// automatically
+	// deleted in [RFC3339](https://www.ietf.org/rfc/rfc3339.txt) text
+	// format.
+	ExpireTime string `json:"expireTime,omitempty"`
+
+	// Location: [Output only] The name of the Google Compute
+	// Engine
+	// [zone](/compute/docs/regions-zones/regions-zones#available)
+	// or
+	// [region](/compute/docs/regions-zones/regions-zones#available) in
+	// which
+	// the cluster resides.
+	Location string `json:"location"`
+
+	// NodeIpv4CidrSize: [Output only] The size of the address space on each
+	// node for hosting
+	// containers. This is provisioned from within the
+	// `container_ipv4_cidr`
+	// range. This field will only be set when cluster is in route-based
+	// network
+	// mode.
+	NodeIpv4CidrSize int64 `json:"nodeIpv4CidrSize,omitempty"`
+
+	// NodePools: The node pools associated with this cluster.
+	// This field should not be set if "node_config" or "initial_node_count"
+	// are
+	// specified.
+	// NOTE(hasheddan): node pools are modelled in status only because
+	// management of node pools is handled by the stack-gcp NodePool object.
+	// TODO(hasheddan): determine if we want to reflect node pools in the
+	// cluster status.
+	// NodePools []*NodePoolClusterStatus `json:"nodePools,omitempty"`
```

Comment:
Do we want to include `NodePools` in the status? We should not be managing them here, but it might be nice to show which pools are going to be deleted if you delete a `GKECluster`. Note that though the GCP has `[]*NodePool` we must use `[]*NodePoolClusterStatus` because we have our own representation of a `NodePool` that includes object metadata.